### PR TITLE
Added remaining signature algorithms and fixed bugs.

### PR DESCRIPTION
**Pull request by xvzcf**
Will squash commits before merging.

@christianpaquin can you confirm this PR works with OpenVPN?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -167,69 +166,6 @@ on **macOS**, run:
 	./Configure no-shared darwin64-x86_64-cc
 	make -j
 
-On Ubuntu and macOS, the fork can also be built as a set of shared libraries by specifying `shared` instead of `no-shared` in the above commands; We have used `no-shared` to avoid having to get the libraries in the right place for the runtime linker.
-
-#### Windows
```
Comment (dstebila) [inline README.md:172]: Why are we removing the Windows build instructions? I thought this worked.

File: README.md [hunk 2]
```diff
@@ -88,39 +84,42 @@ The following quantum-safe algorithms from liboqs are supported (assuming they h
...
 - if `<SIG>` has L3 security, the fork provides the method `p384_<SIG>`, which combines `<SIG>` with ECDSA using NIST's P384 curve.
 - if `<SIG>` has L5 security, the fork provides the method `p521_<SIG>`, which combines `<SIG>` with ECDSA using NIST's P521 curve.
 
-#### Example hybrid algorithms
-
-As `Kyber768` is a NIST L3 KEX algorithm, the hybrid `p384_kyber768` is available.
+For example, since `dilithium2` claims L1 security, the hybrids `rsa3072_dilithium2` and `p256_dilithium2` are available.
 
-As `Dilithium2` is a NIST L1 SIG algorithm, the hybrid `p256_dilithium2` is available.
+**N.B.**: Only the L1 signature variants and `rainbowVcclassic` are enabled by default. Should you wish to enable additional variants, consult [the "Code Generation" section of the documentation in the wiki](https://github.com/open-quantum-safe/openssl/wiki/Using-liboqs-algorithms-not-in-the-fork#code-generation).
```
Comment (dstebila) [inline README.md]: Can you move this paragraph up to line 97?

**Review by dstebila — APPROVED** [review]

**Comment by dstebila** [comment]
Also: update line 302 of README.md with the same list of companies in the liboqs README.md.

File: README.md (hunk 2 above)
Comment (baentsch) [reply README.md]: @xvzcf I'm confused: Wasn't the statement yesterday that all sig-combinations can remain in as nothing breaks this way? The logic described above in my eyes is far from ideal: Do we really want to _remove completely_ from the standard build "interesting" signature combinations ("strong" PQ algorithm with "equivalent" classic algorithm)? 
And reading further into the changes: Do I read correctly in the PR that even plain, non-hybrid PQ sig algs get removed with this PR from the list of 'normally available' signature algorithms? I don't particularly like this. At the very least, I'd like to retain Dilithium3 over Dilithium2 (which security-wise is not really suitable anyway). But then again, strong Falcon also got removed -- and not just the hybrid... 
In general, my question is: What's the deciding factor as to which algorithms to keep by default enabled? If it's performance (?) I'm really unhappy with the default list in this PR. @christianpaquin @dstebila : What's your take on this?

Comment (christianpaquin) [reply README.md]: I don't understand either the logic of removing the algorithms. My understanding from the call yesterday was also that we can keep the existing algorithms as is. In particular, I'd like qtesla-p-iii and picnic2 to remain enabled by default.

File: README.md [hunk 3]
```diff
@@ -88,39 +84,42 @@ The following quantum-safe algorithms from liboqs are supported (assuming they h
...
 
 ## Quickstart
 
-The steps below have been confirmed to work on macOS 10.14 (clang 10.0.0), Ubuntu 18.04.1 (gcc-7), and Windows 10 (VS2019 build tools).
```
Comment (christianpaquin) [inline README.md:123]: Why removing this?

**Review by christianpaquin — CHANGES_REQUESTED** [review]
This PR is quite massive, especially at a RC stage. I would have expected a simple addition of new algs, without removing existing ones (breaking the RC expectations).  Isn't that possible?

File: README.md (hunk 2 above)
Comment (xvzcf) [reply README.md]: I apologize for the confusion. I reverted these changes, but if anyone's puzzled about why I made them in the first place, I could go over my thought process in a call.

File: README.md (hunk 3 above)
Comment (xvzcf) [reply README.md:123]: I made this change before you tried to build it on Windows again, and I forgot to revert it. I've added the instructions back.

**Comment by xvzcf** [comment]
> This PR is quite massive, especially at a RC stage. I would have expected a simple addition of new algs, without removing existing ones (breaking the RC expectations). Isn't that possible?

I've kept the existing ones in. The only difference is that some signature code points have changed (in order to keep everything sequential). If that causes problems with OpenVPN, I can revert those changes too.

**Comment by christianpaquin** [comment]
> > This PR is quite massive, especially at a RC stage. I would have expected a simple addition of new algs, without removing existing ones (breaking the RC expectations). Isn't that possible?
> 
> I've kept the existing ones in. The only difference is that some signature code points have changed (in order to keep everything sequential). If that causes problems with OpenVPN, I can revert those changes too.

That should be ok; I'll ask for a re-test when we tag another RC.

**Review by christianpaquin — APPROVED** [review]

**Pull request merged**